ui/components/src/test/inputs/QuantityInput.test.tsx

```diff
@@ -0,0 +1,89 @@
+    await MockRender.App.startup();
+  });
+
+  after(async () => {
+    await MockRender.App.shutdown();
+    TestUtils.terminateUiComponents();
+    Object.defineProperty(IModelApp, "requestNextAnimation", rnaDescriptorToRestore);
+  });
+
+  afterEach(cleanup);
+
+  it("should render input for Length", async () => {
+    const initialLength = 1;  // 1 meter
+    const spyOnChange = sinon.spy();
+    const wrapper = render(<QuantityInput initialValue={initialLength} quantityType={QuantityType.Length} onQuantityChange={spyOnChange} />);
+    expect(wrapper).not.to.be.undefined;
+    const input = await wrapper.findByTestId("components-parsed-input");
+    fireEvent.change(input, { target: { value: "2.5" } });
+    expect(spyOnChange).not.to.have.been.called;
+    fireEvent.keyDown(input, { key: SpecialKey.Enter });
+    await TestUtils.flushAsyncOperations();
+    expect(spyOnChange).to.have.been.called;
+  });
+
+  it("should process ESC key", async () => {
+    const initialLength = 1;  // 1 meter
+    const spyOnChange = sinon.spy();
+
+    const wrapper = render(<QuantityInput initialValue={initialLength} quantityType={QuantityType.Length} onQuantityChange={spyOnChange} />);
+    expect(wrapper).not.to.be.undefined;
+
+    const input = (await wrapper.findByTestId("components-parsed-input")) as HTMLInputElement;
+    const initialValue = input.value;
+    fireEvent.change(input, { target: { value: "2.5" } });
+    fireEvent.keyDown(input, { key: SpecialKey.Escape });
+    fireEvent.keyDown(input, { key: SpecialKey.Enter });
+    await TestUtils.flushAsyncOperations();
+    expect(spyOnChange).not.to.have.been.called;  // value did not change after ESC was pressed
+    const currentValue = input.value;
+    expect(initialValue).to.eq(currentValue);
+
+    IModelApp.quantityFormatter.onActiveUnitSystemChanged.emit({ useImperial: true });
+  });
+
+  it("should attach 'components-parsed-input-has-error' when bad input", async () => {
+    const initialLength = 1;  // 1 meter
+    const spyOnChange = sinon.spy();
+
+    const wrapper = render(<QuantityInput initialValue={initialLength} quantityType={QuantityType.Length} onQuantityChange={spyOnChange} />);
+    expect(wrapper).not.to.be.undefined;
+    const input = (await wrapper.findByTestId("components-parsed-input")) as HTMLInputElement;
+    const initialValue = input.value;
+    input.focus();
+    fireEvent.change(input, { target: { value: "abc" } });
+    input.blur();
+    await TestUtils.flushAsyncOperations();
+    expect(input.classList.contains("components-parsed-input-has-error")).to.be.true;
+    fireEvent.keyDown(input, { key: SpecialKey.Escape });
+    expect(spyOnChange).not.to.have.been.called;  // value did not change after ESC was pressed
+    const currentValue = input.value;
+    await TestUtils.flushAsyncOperations();
+    expect(input.classList.contains("components-parsed-input-has-error")).to.be.false;
+    expect(initialValue).to.eq(currentValue);
+
+  });

```

Comment:
Make test synchronous by removing `flushAsyncOperations` and replacing `findBy` with `getBy` queries.